lib/opensearch-config/node-config.ts

```diff
@@ -66,86 +66,114 @@ export enum arm64Ec2InstanceType {
+  hasInternalStoage: boolean;
+}
 
-export const getX64InstanceTypes = (instanceType: string) => {
+export const getX64InstanceTypes = (instanceType: string): InstanceTypeInfo => {
   switch (instanceType) {
   case x64Ec2InstanceType.M5_XLARGE:
-    return InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE);
+    return { instance: InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE), hasInternalStoage: false };
   case x64Ec2InstanceType.M5_2XLARGE:
-    return InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE2);
+    return { instance: InstanceType.of(InstanceClass.M5, InstanceSize.XLARGE2), hasInternalStoage: false };
   case x64Ec2InstanceType.C5_LARGE:
-    return InstanceType.of(InstanceClass.C5, InstanceSize.LARGE);
+    return { instance: InstanceType.of(InstanceClass.C5, InstanceSize.LARGE), hasInternalStoage: false };
   case x64Ec2InstanceType.C5_XLARGE:
-    return InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE);
+    return { instance: InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE), hasInternalStoage: false };
   case x64Ec2InstanceType.C5_2XLARGE:
-    return InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE2);
+    return { instance: InstanceType.of(InstanceClass.C5, InstanceSize.XLARGE2), hasInternalStoage: false };
   case x64Ec2InstanceType.R5_LARGE:
-    return InstanceType.of(InstanceClass.R5, InstanceSize.LARGE);
+    return { instance: InstanceType.of(InstanceClass.R5, InstanceSize.LARGE), hasInternalStoage: false };
   case x64Ec2InstanceType.R5_XLARGE:
-    return InstanceType.of(InstanceClass.R5, InstanceSize.XLARGE);
+    return { instance: InstanceType.of(InstanceClass.R5, InstanceSize.XLARGE), hasInternalStoage: false };
   case x64Ec2InstanceType.R5_2XLARGE:
-    return InstanceType.of(InstanceClass.R5, InstanceSize.XLARGE2);
+    return { instance: InstanceType.of(InstanceClass.R5, InstanceSize.XLARGE), hasInternalStoage: false };
   case x64Ec2InstanceType.R5_4XLARGE:
-    return InstanceType.of(InstanceClass.R5, InstanceSize.XLARGE4);
+    return { instance: InstanceType.of(InstanceClass.R5, InstanceSize.XLARGE4), hasInternalStoage: false };
   case x64Ec2InstanceType.R5_8XLARGE:
-    return InstanceType.of(InstanceClass.R5, InstanceSize.XLARGE8);
+    return { instance: InstanceType.of(InstanceClass.R5, InstanceSize.XLARGE8), hasInternalStoage: false };
   case x64Ec2InstanceType.G5_LARGE:
-    return InstanceType.of(InstanceClass.G5, InstanceSize.LARGE);
+    return { instance: InstanceType.of(InstanceClass.G5, InstanceSize.LARGE), hasInternalStoage: true };
   case x64Ec2InstanceType.G5_XLARGE:
-    return InstanceType.of(InstanceClass.G5, InstanceSize.XLARGE);
+    return { instance: InstanceType.of(InstanceClass.G5, InstanceSize.XLARGE), hasInternalStoage: true };
   case x64Ec2InstanceType.I3_LARGE:
-    return InstanceType.of(InstanceClass.I3, InstanceSize.LARGE);
+    return { instance: InstanceType.of(InstanceClass.I3, InstanceSize.LARGE), hasInternalStoage: true };
```

Comment:
How are we maintaining this list? I mean how do we know if the internal storage is available for not? Is it manual or should there be some conditional check that attach if available?

Reply:
Check `node-config.ts`. The method returns whether the instance type supports instance storage or not. 
This cannot be verified dynamically in cdk code.

Reply:
The check is in below method added to this PR
```
 private checkInstanceStorageSettings(hasInstanceStorage: boolean) {
    if (this.useInstanceBasedStorage && !hasInstanceStorage) {
      throw new Error('The instance type provided does not have instance backed storage. Please provide correct instance type');
    }
  }
```

Reply:
Thanks! Yeah I thought there might be a way to retrieve it dynamically instead of maintaining list.